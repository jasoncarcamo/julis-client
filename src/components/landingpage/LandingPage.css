
/*main section*/
#homepage{
    display: inline-block;
    width: 100%;
    padding: 0;
    margin-top: 15%;
}


/*first section*/
#first_section{    
    background-image: url("../../picture-assests/first_section.jpg");
    background-repeat: no-repeat;
    background-size: cover;
    width: 100%;
    height: 75vh;
    margin: 0;
    padding: 0;
    border: 1px solid lightgrey;
}

#first_section > header > h1{
    font-size: 1.3em;
    font-weight: bold;
    margin-left: 5%;
    padding: 1.5%;
}

#first_section > h2{
    font-size: 1.6em;
    margin: 15% 0 -2% 0;    
    text-align: center;
    color: white;
}

#first_section p{
    width: 60%;
    margin: 3% auto;
    font-size: 1.1em;
    text-align: center;
    color: white;
}

#first_section a{
    display: block;
    width: 9em;
    height: 5em;
    border: 1px solid white !important;
    margin: 4% auto 0 auto;
    text-align: center;
    padding-top: 3%;
    border: none;
    font-size: 1.1em;
    color: white;
    text-decoration: none;
}


/*features section*/
#features_section{
    display: grid;
    width: 100%;
    grid-template: 1fr / auto auto auto;
    grid-gap: 2em;
    padding: 5% 7%;
    margin: 2% 0;
    background-color: rgb(218, 218, 218);
}

#features_section section{ 
    padding: 0;
    text-align: center;
    background-color: white;
}

#features_section section p{
    width: 80%;
    margin: 10% auto 4% auto;
    text-align: center;
}

#child_friendly img{
    width: 100%;
    height: 50%;
}

#clean_kitchen p{
    width: 90%;
}

#clean_kitchen img {
    width: 100%;
    height: 15em;
}

#clean_office img{
    width: 100%;
    height: 50%;
}

/* living room section */
#living_room_section{
    display: grid;
    grid-template: 1fr / 1fr 1fr;
}

#clean_living_room{
    width: 100%;
    height: 25em;
}

#living_room_section p{
    width: 95%;
    height: 50%;
    text-align: center;
    margin: auto;
}

.to_signup{
    display: block;
    text-align: center;
    margin: 7% auto 0 auto;
    padding-top: 1.5%;
    width: 8em; 
    height: 3.5em;
    border: 1px solid black;
    text-decoration: none;
    color: black;
}



@media screen and (max-width: 445px ){
  #homepage{
      margin-top: 25%;
  }
/*first section*/  

#first_section{
    height: 25em;
}

    #first_section a{
        border: 1px solid black;
        margin-top: 10%;
        padding-top: 6%;
    }

    /*features section*/
    #features_section{
        grid-template: auto auto auto / 1fr;
    }

    #features_section section{
        width: 75%;
        height: auto;
        margin: 0 auto;
    }


    /*living room section */
    #living_room_section{
        grid-template: 1fr 1fr / 1fr;
    }

    #clean_living_room{
        display: block;
        height: auto;
    }

    #living_room_section p{
        margin: 0 auto  ;
    }

    .to_signup{
        padding-top: 3%;
        margin-top: 30% !important;
    }
}

@media screen and (max-width: 768px){

    #first_section{
        height: 25em;
    }
    #features_section section img{
        height: 25%;
    }

    #living_room_section p{
        margin: auto;
    }

    .to_signup{
        margin-top: 10%;
    }
}