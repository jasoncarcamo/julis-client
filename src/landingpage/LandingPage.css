#hello{
    height: 50vh;
    margin: 2% 0;
    background-color: black; 
}