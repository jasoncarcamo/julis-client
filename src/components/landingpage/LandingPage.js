import React from 'react';
import './LandingPage.css';
import cleanCrib from '../../picture-assests/clean_crib.jpg';
import cleanKitchen from '../../picture-assests/clean_kitchen.jpg';
import cleanOffice from '../../picture-assests/clean_office.jpg';
import cleanLivingRoom from '../../picture-assests/clean_living_room.jpg';
import {Link} from 'react-router-dom';



class LandingPage extends React.Component{
    render(){
        return (
            <section id="homepage">

                <section id="first_section">
                    
                    <header><h1>Julis Cleaning Service</h1></header>
                    <h2>Quality Service at an Affordable Price</h2>
                    <p>Commercial & Industrial Sanitation Professionals</p>
                    <a href="#child_friendly">About Us</a>
                </section>

                <section id="features_section">

                    <section id="child_friendly">   

                        <img src={cleanCrib} alt="A clean baby crib."/>
                        <h3>Green Friendly</h3>
                        <p>We use environmentally-conscious products that are safe for people and pets.</p>

                    </section>

                    <section id="clean_kitchen">

                        <img src={cleanKitchen} alt="A clean kitchen"/>
                        <h3>Residential and Commercial cleaning</h3>
                        <p>From deep cleaning your kitchen and bathrooms to changing your bed linens, we are dedicated to creating clean, safe and healthy environments for both commercial and residential spaces.</p>

                    </section>

                    <section id="clean_office">

                        <img src={cleanOffice} alt="A clean office"/>
                        <h3>Customer satisfaction</h3>
                        <p>Whether you need weekly, every other week, monthly, or occasional housekeeping services, our professionals will leave you feeling comfortable in and proud of your space.</p>

                    </section>

                </section>

                <section id="living_room_section">
                    <img id="clean_living_room" src={cleanLivingRoom} alt="A clean living room."/>
                    <p>Coming home after a long day to a place that’s clean, orderly, and welcoming does not have to be a fantasy. That’s why we take the time to offer services that work with your budget, cleaning preferences, and busy lifestyle.  Our friendly professionals strive to earn your loyalty every time we clean, which is why we make setting up house cleaning services as easy as can be. Whether you are looking to request services one time or create a routine cleaning plan, you'll never have to worry about signing a contract.</p>

                </section>

                <section id="landing_sign_up">
                    <p></p>
                    <Link to="/register" className="to_signup">Get Started</Link>
                </section>

            </section>
        )
    }
}

export default LandingPage;