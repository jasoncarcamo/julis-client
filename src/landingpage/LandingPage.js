import React from 'react';
import './LandingPage.css';


class LandingPage extends React.Component{
    render(){
        return (
            <div id="hello"></div>
        )
    }
}

export default LandingPage;