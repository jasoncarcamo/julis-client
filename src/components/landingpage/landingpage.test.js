import React from "react";
import ReactDOM from "react-dom";
import LandingPage from "./LandingPage";
import {BrowserRouter} from "react-router-dom";
import Enzyme from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter()});


describe("<LandingPage/>", ()=>{
    it("Renders landing page compeonent", ()=>{
        const div = document.createElement("div");

        ReactDOM.render(<BrowserRouter><LandingPage/></BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("create snapshot for landing page component", ()=>{
        const tree = renderer.create(<BrowserRouter><LandingPage/></BrowserRouter>).toJSON();

        expect(tree).toMatchSnapshot();
    })

})