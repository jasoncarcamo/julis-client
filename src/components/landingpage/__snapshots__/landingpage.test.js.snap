// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingPage/> create snapshot for landing page component 1`] = `
<section
  id="homepage"
>
  <section
    id="first_section"
  >
    <header>
      <h1>
        Julis Cleaning Service
      </h1>
    </header>
    <h2>
      Quality Service at an Affordable Price
    </h2>
    <p>
      Commercial & Industrial Sanitation Professionals
    </p>
    <a
      href="#child_friendly"
    >
      About Us
    </a>
  </section>
  <section
    id="features_section"
  >
    <section
      id="child_friendly"
    >
      <img
        alt="A clean baby crib."
        src="clean_crib.jpg"
      />
      <h3>
        Green Friendly
      </h3>
      <p>
        We use environmentally-conscious products that are safe for people and pets.
      </p>
    </section>
    <section
      id="clean_kitchen"
    >
      <img
        alt="A clean kitchen"
        src="clean_kitchen.jpg"
      />
      <h3>
        Residential and Commercial cleaning
      </h3>
      <p>
        From deep cleaning your kitchen and bathrooms to changing your bed linens, we are dedicated to creating clean, safe and healthy environments for both commercial and residential spaces.
      </p>
    </section>
    <section
      id="clean_office"
    >
      <img
        alt="A clean office"
        src="clean_office.jpg"
      />
      <h3>
        Customer satisfaction
      </h3>
      <p>
        Whether you need weekly, every other week, monthly, or occasional housekeeping services, our professionals will leave you feeling comfortable in and proud of your space.
      </p>
    </section>
  </section>
  <section
    id="living_room_section"
  >
    <img
      alt="A clean living room."
      id="clean_living_room"
      src="clean_living_room.jpg"
    />
    <span>
      <p>
        Coming home after a long day to a place that’s clean, orderly, and welcoming does not have to be a fantasy. That’s why we take the time to offer services that work with your budget, cleaning preferences, and busy lifestyle.  
      </p>
      <p>
        Our friendly professionals strive to earn your loyalty every time we clean, which is why we make setting up house cleaning services as easy as can be. Whether you are looking to request services one time or create a routine cleaning plan, you'll never have to worry about signing a contract.
      </p>
    </span>
  </section>
  <section
    id="landing_sign_up"
  >
    <a
      className="to_signup"
      href="/register"
      onClick={[Function]}
    >
      Get Started
    </a>
  </section>
</section>
`;
